make/common/JavaCompilation.gmk

```diff
@@ -437,26 +438,26 @@ define SetupJavaCompilationBody
       $1_API_DIGEST_FLAGS := \
           -classpath $$(BUILDTOOLS_OUTPUTDIR)/depend \
           -Xplugin:"depend $$($1_API_TARGET)" \
+          "-XDinternalAPIPath=$$($1_API_INTERNAL)" \
           #
 
       $1_EXTRA_DEPS := $$(BUILDTOOLS_OUTPUTDIR)/depend/_the.COMPILE_DEPEND_batch
     endif
 
-    # Create a file with all sources, to pass to javac in an @file.
+    # Do the actual compilation
+    # Creates a file with all sources, to pass it to javac in an @file.
     # $$($1_VARDEPS_FILE) is used as dependency to track changes in set of
     # list of files.
-    $$($1_FILELIST): $$($1_SRCS) $$($1_VARDEPS_FILE)
-		$$(call MakeDir, $$(@D))
-		$$(call LogWarn, Compiling $$(words $$($1_SRCS)) files for $1)
-		$$(eval $$(call ListPathsSafely, $1_SRCS, $$($1_FILELIST)))
-
-    # Do the actual compilation
-    $$($1_COMPILE_TARGET): $$($1_SRCS) $$($1_FILELIST) $$($1_DEPENDS) \
+    $$($1_COMPILE_TARGET): $$($1_SRCS) $$($1_DEPENDS) \
```

Comment:
What's the reason for inlining this recipe into the main compilation recipe? I would think generating the modfiles file in the FILELIST rule would help reduce the noise from non source files in the prerequisites list (as the old FILELIST rule had a lot less prerequisites listed). Keeping this recipe separate also limits regeneration of the FILELIST.

Reply:
I've tried to fix this in:
https://github.com/openjdk/jdk/pull/10104/commits/f1e19e8b636ddf08500a6401ce1488265f7b9446